### PR TITLE
Add several tests

### PR DESCRIPTION
**Pull request by martindurant**
Introduced FileNotFoundError with ls on a bucket that cannot be accessed

### CODE REVIEW

File: s3fs/tests/test_s3fs.py [hunk 1]
```diff
@@ -113,12 +132,19 @@ def test_rm(s3):
...
+    assert not s3.exists(test_bucket_name+'/nested/nested2/file1')
 
 
 def test_anonymous_access():
     with ignoring(NoCredentialsError):
-        fs = S3FileSystem(anon=True)
-        fs.ls(test_bucket_name)
+        s3 = S3FileSystem(anon=True)
+        assert s3.ls('') == []
+        ## TODO: public bucket doesn't work through moto
```
Comment (mrocklin) [inline s3fs/tests/test_s3fs.py]: Is this TODO still the case with the `ACL` change above?

**Comment by mrocklin** [comment]
This looks fantastic to me.

File: s3fs/tests/test_s3fs.py (hunk 1 above)
Comment (martindurant) [inline s3fs/tests/test_s3fs.py]: ACL did not seem to fix this. I'll look into it, but intend to leave as-is for now.

**Comment by mrocklin** [comment]
Is it safe to delete these extra branches hanging out on the main fork?

**Pull request merged**